docs/ocs/troubleshooting.md

```diff
@@ -0,0 +1,169 @@
+
+```
+time="2021-11-23T17:07:41-05:00" level=info msg="CRD Status:"
+time="2021-11-23T17:07:41-05:00" level=info msg="✅ Exists: CustomResourceDefinition \"noobaas.noobaa.io\"\n"
+[...]
+time="2021-11-23T17:07:41-05:00" level=info msg="Operator Status:"
+time="2021-11-23T17:07:41-05:00" level=info msg="✅ Exists: Namespace \"openshift-storage\"\n"
+time="2021-11-23T17:07:41-05:00" level=info msg="✅ Exists: ServiceAccount \"noobaa\"\n"
+time="2021-11-23T17:07:41-05:00" level=info msg="✅ Exists: ServiceAccount \"noobaa-endpoint\"\n"
+[...]
+time="2021-11-23T17:07:41-05:00" level=info msg="System Wait Ready:"
+time="2021-11-23T17:07:41-05:00" level=info msg="✅ System Phase is \"Ready\".\n"
+time="2021-11-23T17:07:41-05:00" level=info
+time="2021-11-23T17:07:41-05:00" level=info
+time="2021-11-23T17:07:41-05:00" level=info msg="System Status:"
+time="2021-11-23T17:07:41-05:00" level=info msg="✅ Exists: NooBaa \"noobaa\"\n"
+time="2021-11-23T17:07:41-05:00" level=info msg="✅ Exists: StatefulSet \"noobaa-core\"\n"
+[...]
+time="2021-11-23T17:07:42-05:00" level=info msg="✅ System Phase is \"Ready\"\n"
+time="2021-11-23T17:07:42-05:00" level=info msg="✅ Exists:  \"noobaa-admin\"\n"
+```
+
+The status information includes various configuration details,
+including the external URL for the Noobaa management interface. You
+should be able to log into that UI using your OpenShift credentials.
+
+## Troubleshooting Ceph connectivity
+
+You can log into the `rook-ceph-tools` container to run Ceph commands
+to diagnose Ceph cluster health and connectivity problems. To log into
+the pod:
+
+```
+oc -n openshift-storage rsh deploy/rook-ceph-tools
+```
+
+If this deployment isn't available, you can enable it by [editing the
+`OCSInitialization`
+object](https://access.redhat.com/articles/4628891).
+
+### Configure credentials
+
+You will
+first need to make sure that appropriate credentials exist in
+`/etc/ceph`. Look at `/etc/ceph/keyring`; if it only contains...
+
+```
+[client.admin]
+key = admin-secret
+```
+
+...then you will need to add credentials for the user we use to
+authenticate to the ceph cluster.  You can obtain the appropriate
+credentials by running:
+
+
+```
+oc -n openshift-storage get secret rook-ceph-external-cluster-details -o json |
+  jq -r '.data.external_cluster_details|@base64d' |
+  jq '.[]|select(.name == "rook-csi-rbd-provisioner")|.data'
+```
+
+This will return a JSON block that looks like:
+
+```
+{
+  "userID": "...",
+  "userKey": "..."
+}
+```
+
+Replace the contents of `/etc/ceph/keyring` with:
+
+```
+[client.<value from userID>]
+key = <value from userKey>
+```
+
+### Check ceph status
+
+Run the `ceph health` command:
+
+```
+$ ceph --id provisioner-moc-rbd-1 health
+HEALTH_WARN 5 clients failing to respond to cache pressure; 12 nearfull osd(s); 1 pool(s) full; 23 pool(s) nearfull; 1 pgs not deep-scrubbed in time; 19 pgs not scrubbed in time
+```
+
+What we care about here is that we get a valid response back from the
```

Comment:
What is a valid response in this case, or what is not a valid response?

Reply:
A valid response is anything that returns ceph server status.